src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx

```diff
@@ -419,10 +448,45 @@ class FilterEditorComponent extends Component<FilterEditorProps, State> {
   private onCustomLabelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     const customLabel = event.target.value;
     this.setState({ customLabel });
+    if (this.props.onLocalFilterUpdate) {
+      if (this.state.isCustomEditorOpen) {
+        const { queryDsl } = this.state;
+        this.props.onLocalFilterUpdate({ queryDsl, customLabel });
+      } else {
+        const localFilter = {
+          ...this.state.localFilter,
+          meta: {
+            ...this.state.localFilter.meta,
+            alias: customLabel || null,
+          },
+        };
+        this.props.onLocalFilterUpdate(localFilter);
+      }
+    }
   };
 
   private onQueryDslChange = (queryDsl: string) => {
     this.setState({ queryDsl });
+    if (this.props.onLocalFilterUpdate) {
+      const { customLabel } = this.state;
+      this.props.onLocalFilterUpdate({ queryDsl, customLabel });
+    }
+  };
+
+  private getFilterFromQueryDsl = (queryDsl: string) => {
+    const { customLabel } = this.state;
+    const {
+      $state,
+      meta: { index, disabled = false, negate = false },
+    } = this.props.filter;
+
+    if (!$state || !$state.store) {
+      return;
+    }
+
+    const newIndex = index || this.props.indexPatterns[0].id!;
+    const body = JSON.parse(queryDsl);
```

Comment:
You wrapped this parse above with a try/catch block, should we do that also here?